sdks/universal-router-sdk/src/entities/actions/uniswap.ts

```diff
@@ -79,24 +78,51 @@ export class UniswapTrade implements Command {
   }
 
   get inputRequiresWrap(): boolean {
-    if (!this.isAllV4) {
-      return this.trade.inputAmount.currency.isNative
+    if (this.isAllV4) {
+      return (
+        this.trade.inputAmount.currency.isNative &&
+        !(this.trade.swaps[0].route as unknown as V4Route<Currency, Currency>).pathInput.isNative
+      )
     } else {
-      // We only support wrapping all ETH or no ETH currently. We cannot support splitting where half needs to be wrapped
-      // If the input currency is ETH and the input of the first path is not ETH it must be WETH that needs wrapping
-      return this.trade.inputAmount.currency.isNative && !this.trade.swaps[0].route.input.isNative
+      return this.trade.inputAmount.currency.isNative
```

Comment:
just wondering - by removing ` && !this.trade.swaps[0].route.input.isNative`, does it mean we can handle ETH/WETH split input at ur-sdk level now?

Reply:
it's technically still there, the conditionals just switched places, this second one, is the first one now. I can add a comment somewhere, but it applies to 4 functions now instead of just 1

Reply:
right now all swaps must have 1 single input token and 1 single output token always.